forward43/scrape_some_good.py

```diff
@@ -0,0 +1,65 @@
+import json
+import requests
+from bs4 import BeautifulSoup
+import pandas as pd
+
+# create url's for ... overview pages
+browse_url_base        = 'https://startsomegood.com/projects?page='
+browse_pages = []
+
+for page in range(15):
+    browse_pages.append(browse_url_base + str(page))
+
+# create dataframe to store data
+column_names            = ["Title", "Description", "Status", "innovation_type", "Country", "City", "Contact person/details", "Link"]
+good_data               = []
+
+# loop over the overview pages to find the project tiles
+project_url_base        = 'https://startsomegood.com'
+
+for URL in browse_pages:
+    RESPONSE_OVERVIEW = requests.get(URL, timeout = 5)
+    SOUP_OVERVIEW     = BeautifulSoup(RESPONSE_OVERVIEW.content, "html.parser")
+
+    PROJECTS = SOUP_OVERVIEW.find_all('a', class_="col-sm-6 col-md-4 project-tile project-tile-link")
+    
+    # loop over the tiles to find the data needed
+    for project in PROJECTS:
+        # Get available data from project tile on overview page
+        title               = project.find('h4').text
+        status              = project.find('span', class_='pull-right closing-date').text
+        link                = project_url_base + project.get('href')
+        
+        # Scrape soup from project page
+        project_response    = requests.get(link, timeout = 5)
+        project_soup        = BeautifulSoup(project_response.content, "html.parser")
+        
+        # Get remaining data from project-page soup
+        if project_soup.find("div", class_="story") is None:
+            description     = 'Description not available'
```

Comment:
I see the content is quite in the webpage is quite messy. Nice that you got this working. 

I think we can simplify the code here a bit. Python's `defaultdict` offers a nice functionality of assigning default strings. See the code below
```
        row = defaultdict(lambda: 'n.a.')

        story = project_soup.find('p', class_='story')
        if story is not None:
            row['description'] = story.text

        subtitle = project_soup.find("p", class_="sub-title")
        if subtitle is not None:
            row['location'] = subtitle.text.strip()
            row['city']     = row['location'].split(',')[0]
            row['country']  = row['location']
            row['contact']  = row['location']
```
The cool thing is, for example even if `description` is not set, `row['description']` will give you the value `n.a.`